activation/activation.go

```diff
@@ -433,6 +442,31 @@ func (b *Builder) SetMinGas(value uint64) {
 	panic("not implemented")
 }
 
+// EstimateReward try to estimate the reward for the given layer.
+func (b *Builder) EstimateReward(layerID uint32) (amount uint64, err error) {
+	if layerID < b.layerClock.GetCurrentLayer().Value {
+		// If layer is in the past, we can just serve value from `rewards` table.
+		return b.cdb.GetRewardForLayer(layerID)
+	}
+	if b.pendingATX == nil {
+		return 0, errors.New("no pending ATX")
+	}
+	// try to calculate possible reward for layer.
+	layerReward := b.rewardConf.CalculateLayerReward(layerID)
+	totalWeight, activeSet, err := b.cdb.GetEpochWeight(types.NewLayerID(layerID).GetEpoch())
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to get epoch weight")
+	}
+	if len(activeSet) == 0 {
+		return 0, errors.New("empty active set for epoch")
+	}
+	expNumSlots, err := proposals.GetNumEligibleSlots(b.pendingATX.GetWeight(), totalWeight, uint32(b.layerAvgSize), b.layersPerEpoch)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to estimate number of eligible slots")
+	}
+	return layerReward / uint64(expNumSlots), nil
```

Comment:
this doesn't make sense

layer reward is divided between builders of the block, based on their relative weight

Reply:
`layerReward` is just basic reward, which can be get for layer. Now it hardcoded value, but later it will be replaces with inflation formula

so we need manually divided to all block builders